### ISSUE TITLE
Errors when following the init instructions

### ISSUE THREAD

**Issue by mstahl** [microsoft/react-native-windows#11909]
### Problem Description

I just created a new React Native app using the instructions provided, with the command `npx react-native init hello_windows --template "react-native@^0.72.0"`, which succeeded, but when I run the next command, `npx react-native-windows-init --overwrite`, the command fails with the following output:

```text
{
  _: [
    '/home/max/.nvm/versions/node/v18.17.0/bin/node',
    '/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/.bin/react-native-windows-init'
  ],
  overwrite: true,
  verbose: true,
  telemetry: true,
  language: 'cpp',
  projectType: 'app',
  'project-type': 'app',
  experimentalNuGetDependency: false,
  'experimental-nu-get-dependency': false,
  useHermes: false,
  'use-hermes': false,
  useWinUI3: false,
  'use-win-u-i3': false,
  useDevMode: undefined,
  'use-dev-mode': undefined,
  '$0': '../../.npm/_npx/966c6a96be6f5a32/node_modules/.bin/react-native-windows-init'
}
/bin/sh: 1: undefinedSystem32reg.exe: not found
dir: cannot access '/-C': No such file or directory
Reading project name from package.json...
Reading react-native version from node_modules...
Reading react-native version from node_modules...
Latest matching version of react-native-windows for react-native@0.72.3 is react-native-windows@0.72.1
Installing react-native-windows@0.72.1...
yarn install v1.22.15
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.33s.
react-native-windows@0.72.1 successfully installed.
Failure during generate  Error: Cannot find module '/home/max/work/hello_windows/node_modules/@react-native/codegen/lib/schemaValidator'
Require stack:
- /home/max/work/hello_windows/node_modules/@react-native-windows/codegen/lib-commonjs/index.js
- /home/max/work/hello_windows/node_modules/@react-native-windows/cli/lib-commonjs/codegen.js
- /home/max/work/hello_windows/node_modules/@react-native-windows/cli/lib-commonjs/index.js
- /home/max/work/hello_windows/node_modules/react-native-windows/generate.js
- /home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/lib-commonjs/requireGenerateWindows.js
- /home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/lib-commonjs/Cli.js
- /home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/bin.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Function.Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at Module.patchedRequire (/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/diagnostic-channel/dist/src/patchRequire.js:16:46)
    at Module.Hook._require.Module.require (/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/require-in-the-middle/index.js:167:34)
    at Module.Hook._require.Module.require (/home/max/work/hello_windows/node_modules/require-in-the-middle/index.js:167:34)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/home/max/work/hello_windows/node_modules/@react-native-windows/codegen/src/index.ts:38:25)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1310:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/max/work/hello_windows/node_modules/@react-native-windows/codegen/lib-commonjs/index.js',
    '/home/max/work/hello_windows/node_modules/@react-native-windows/cli/lib-commonjs/codegen.js',
    '/home/max/work/hello_windows/node_modules/@react-native-windows/cli/lib-commonjs/index.js',
    '/home/max/work/hello_windows/node_modules/react-native-windows/generate.js',
    '/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/lib-commonjs/requireGenerateWindows.js',
    '/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/lib-commonjs/Cli.js',
    '/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/bin.js'
  ]
}
generateWindows is not a function
TypeError: generateWindows is not a function
    at Object.reactNativeWindowsInit (/home/max/.npm/_npx/966c6a96be6f5a32/node_modules/react-native-windows-init/src/Cli.ts:626:11)
Command failed with error Unknown: generateWindows is not a function
Your telemetry sessionId was 5eb6644488d007d7b2e11a986294f3f3
```

I took a look at that node module in the `node_modules` directory and that file, `schemaValidator`, does exist but it's called `SchemaValidator`. So, this might be the result of a typo or maybe something wilder than that.

### Steps To Reproduce

1. Open Windows Subsystem for Linux. I'm using Ubuntu in WSL.
2. Run the command `npx react-native init hello_windows --template "react-native@^0.72.0"`
3. Run the command `cd hello_windows`
4. Run the command `npx react-native-windows-init --overwrite --verbose` to get the output I pasted above.

### Expected Results

I would've expected the command to finish without errors, and for the project to then have a `windows/` directory in it with Visual Studio stuff in it or whatever's supposed to happen.

### CLI version

11.3.5

### Environment

```markdown
info Fetching system and libraries information...
System:
  OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
  CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
  Memory: 30.36 GB / 31.31 GB
  Shell:
    version: "5.8"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.0
    path: ~/.nvm/versions/node/v18.17.0/bin/node
  Yarn:
    version: 1.22.15
    path: ~/.yarn/bin/yarn
  npm:
    version: 9.8.0
    path: ~/.nvm/versions/node/v18.17.0/bin/npm
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
Languages:
  Java: Not Found
  Ruby:
    version: 2.7.2
    path: /home/max/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.3
    wanted: 0.72.3
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

Literally haven't written any code. This is just me trying to get started in the first place initializing a fresh project.

**Comment by chrisglein** [comment]
This sounds like the combination of causes:

> I took a look at that node module in the node_modules directory and that file, schemaValidator, does exist but it's called SchemaValidator. So, this might be the result of a typo or maybe something wilder than that.

> Open Windows Subsystem for Linux. I'm using Ubuntu in WSL.

Windows definitely has case insensitivity that WSL isn't going to.

It certainly seems like there's a difference here: https://github.com/microsoft/react-native-windows/blob/a66d1c6238f1f330b365cb7c392b73b3f92e6183/packages/%40react-native-windows/codegen/src/index.ts#L41
```js
const schemaValidator = require(path.resolve(
  rncodegenPath,
  'lib/SchemaValidator',
));
```
And a relevant code change here: https://github.com/microsoft/react-native-windows/commit/0e9e464c8b9f73e941e502efcce11be1aeb5b663 ("Fix RNW not being detected on case-sensitive file systems")
@tido64, Is this expected to be fixed?

**Comment by tido64** [comment]
> @tido64, Is this expected to be fixed?

The fix was cherry-picked to 0.71, and was expected to be included in 0.72 but I guess it missed the window. We should cherry-pick it to 0.72.

**Comment by jonthysell** [comment]
Should also mention that running RNW CLI commands in WSL is not a tested scenario - there may be places where we explicitly fail because we expect to be in Windows.

Also, the getting started docs were slightly out of date (you should be using RN >= 0.72.3 and RNW >= 0.72.2). PR to fix the docs here: https://github.com/microsoft/react-native-windows-samples/pull/871

**Comment by tido64** [comment]
@mstahl This should be fixed as of 0.72.3

**Comment by mstahl** [comment]
> Should also mention that running RNW CLI commands in WSL is not a tested scenario - there may be places where we explicitly fail because we expect to be in Windows.

This turned out to be the issue. It'd be cool to keep it all in WSL but using powershell or cmd works fine for me, so no worries. Thanks y'all!
